oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/LogoutHandler.java

```diff
@@ -231,45 +159,149 @@ private APIGatewayProxyResponseEvent validateClientIDAgainstClientRegistry(
                     new ErrorObject(OAuth2Error.UNAUTHORIZED_CLIENT_CODE, "client not found"),
                     input,
                     Optional.of(clientID),
-                    sessionId);
+                    Optional.empty());
+        }
+        if (postLogoutRedirectUri.isEmpty()) {
+            LOG.info(
+                    "post_logout_redirect_uri is NOT present in logout request. Generating default logout response");
+            return logoutService.generateDefaultLogoutResponse(
+                    state, input, Optional.of(clientID), Optional.empty());
         }
+        if (!checkPostLogoutRedirectUrlInClientReg(postLogoutRedirectUri, clientRegistry)) {
+            return errorLogoutWithSessionCheck(sessionFromSessionCookie, input, state, clientID);
+        }
+
+        if (sessionFromSessionCookie.isPresent()) {
+            return segmentedFunctionCall(
+                    "logoutWithSession",
+                    () ->
+                            logoutWithSession(
+                                    idTokenHint.get(),
+                                    sessionFromSessionCookie.get(),
+                                    clientID,
+                                    postLogoutRedirectUri.get(),
+                                    state,
+                                    input));
+
+        } else {
+            return segmentedFunctionCall(
+                    "generateDefaultLogoutResponse",
+                    () ->
+                            logoutService.generateLogoutResponse(
+                                    URI.create(postLogoutRedirectUri.get()),
+                                    state,
+                                    Optional.empty(),
+                                    input,
+                                    Optional.of(clientID),
+                                    Optional.empty()));
+        }
+    }
 
+    private APIGatewayProxyResponseEvent defaultLogoutWithSessionCheck(
+            Optional<Session> sessionFromSessionCookie,
+            APIGatewayProxyRequestEvent input,
+            Optional<String> state) {
+        Session session;
+        Optional<String> sessionResponse;
+        if (sessionFromSessionCookie.isPresent()) {
+            session = sessionFromSessionCookie.get();
+            segmentedFunctionCall("destroySessions", () -> logoutService.destroySessions(session));
```

Comment:
There is a "Deleting Session" log within the service ? Does that cover what you are saying?

Reply:
look further

Reply:
As discussed, logs are piped through to splunk and are used for various monitoring purposes/can search them based on message etc. Should be looked into whether changing this will have a knock on effect (as with other logs as previously mentioned)

Reply:
Not easy to have the exact previous log message when the same conditions occur. Feels like it would be manipulating the code a bit too much to get that log message again when it might not be needed ?

Reply:
Isaac has checked that the changing of these logs has no impact